ee/powereventwatcher/power_event_watcher_windows.go

```diff
@@ -28,14 +28,31 @@ type (
 
 	powerEventWatcher struct {
 		slogger                 *slog.Logger
-		knapsack                types.Knapsack
+		powerEventSubscriber    powerEventSubscriber
 		subscriptionHandle      uintptr
 		subscribeProcedure      *syscall.LazyProc
 		unsubscribeProcedure    *syscall.LazyProc
 		renderEventLogProcedure *syscall.LazyProc
 		interrupt               chan struct{}
 		interrupted             bool
 	}
+
+	// powerEventSubscriber is an interface to be implemented by anything utilizing the power event updates.
+	// implementers are provided to New, and the interface methods below are called as described during relevant updates
+	powerEventSubscriber interface {
+		// OnPowerEvent will be called for the provided subscriber whenever any watched event is observed
+		OnPowerEvent(eventID int) error
+		// OnStartup will be called when the powerEventWatcher is initially set up, allowing subscribers
+		// to perform any setup behavior (e.g. cache clearing, state resetting)
+		OnStartup() error
+	}
+
+	// knapsackSubscriber implements the powerEventSubscriber interface and
+	// updates the knapsack.InModernStandby state based on the power events observed
+	knapsackSubscriber struct {
```

Comment:
This might just be me, but I was confused because `knapsackSubscriber` sounds like something subscribing to the knapsack and not (effectively) the knapsack subscribing to something else. What about something like `knapsackUpdater` (or something along those lines but more specific, like `knapsackSleepStateUpdater`)?

Reply:
oh yeah, that makes more sense thank you. will do `knapsackSleepStateUpdater`